src/blockloaders.js

```diff
@@ -48,13 +63,7 @@ async function createLoader(opts) {
   }
 
   // custom provided loaders
-  switch (scheme) {
```

Comment:
What if we kept this, and simply had it do this, using `toGatewayURL()` from auto-js-ipfs?

Would then not need to call createIPFS or have IPFSRangeLoader at all?

Thinking this may simplify codebase a bit / remove some redundancy?

```
  // custom provided loaders
  switch (scheme) {
    case "ipfs":
      opts.url = toGatewayURL(opts.url);
      return new FetchRangeLoader(opts);
  }
  ```

Reply:
If we use gateways only, then we'd lose the ability to use kubo and other interfaces in the future. As it is, it could detect a running kubo instance from something like IPFS Companion and use that for actual p2p when possible.

Personally, I'm in favor of leaving it as is so we can build up future work on top of it.

Reply:
Ok, as discussed, let's keep this as is for now, can revisit later once we try the use case in an extension to see if it makes sense. Maybe will simplify in the future.